### PR TITLE
Remove references to cli flags in errors

### LINKED ISSUE #880: Remove References to CLI Flags in Error Messages
s2i is often run in a container and configured via environment variables.
Error messages printed to stderr should not reference command-line flags, as these configuration values can be set by other means.

### PR DESCRIPTION
**Pull request by adambkaplan**
Fixes #880

### CODE REVIEW

File: pkg/errors/errors.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func NewPullImageError(name string, err error) error {
 		Message:    fmt.Sprintf("unable to get %s", name),
 		Details:    err,
 		ErrorCode:  PullImageError,
-		Suggestion: "check image name, or if using local image add --pull-policy=never flag",
+		Suggestion: fmt.Sprintf("check image name, or if using a local image set the builder image pull policy to %q", "never"),
```
Comment (bparees) [inline pkg/errors/errors.go]: is this preferred over escaping quotes in the string? Seems odd.

**Comment by bparees** [comment]
travis is unhappy as a result of my earlier PR:
```
./pkg/scm/downloaders/file/download.go:17:6: exported type RecursiveCopyError should have comment or be unexported
```

can you go ahead and fix it here?

File: pkg/errors/errors.go (hunk 1 above)
Comment (adambkaplan) [reply pkg/errors/errors.go]: %q seems to be a short cut, and I've seen it used elsewhere. Under the hood go does additional cleanup, like trimming whitespace.
If this was a performance-intensive, I wouldn't use it (then again, I'd avoid `fmt` for anything high-performance)

Comment (bparees) [reply pkg/errors/errors.go]: k

**Comment by adambkaplan** [comment]
[test]

**Comment by bparees** [comment]
[merge]

**Pull request merged**